### ISSUE TITLE
Forward/backward mouse side buttons with waynergy

### ISSUE THREAD

**Issue by donjan** [input-leap/input-leap#1492]
### What happened?

I'm toying with connecting a Ubuntu 20.04 barrier server, now switched to a _input-leap_ master checkout in hope of better mouse button support from https://github.com/input-leap/input-leap/pull/391/, with a Ubuntu 22.04 _waynergy_ client (since Wayland seems inevitable...).

Works reasonably so far, but the forward/backward side buttons on my Logitech M500 don't register on the client, while they do on the server. They also work when the mouse is connected to the client machine directly.
`xev` output on the 20.04 host tells me:
```
button 1: LMB
button 2: wheel click
button 3: RMB
button 4: scroll up
button 5: scroll down
button 6: left tilt wheel
button 7: right tilt wheel
button 8: backward side button
button 9: forward side button
```
`xev` on the 22.04 client shows the same when directly wired, but only up to and including button 7 via input-leap. Then buttons 8 and 9 do nothing.
Notably wheel tilting (buttons 6, 7) for horizontal scrolling does work on the client via input-leap.

### Version

v2.4.0

### Git commit hash (if applicable)

708cbfde

### What OSes are you seeing the problem on? (Check all that apply)

Linux

### What OS versions are you using?

Ubuntu 20.04, Ubuntu 22.04

### Any other information

I could work around this by remapping the buttons to lower indices on both machines, e.g. moving the horizontal scrolling to the unsupported range. The side buttons offer higher usability and quality of life for me, but maybe this can be done without sacrifice and a fix in input-leap would be more appropriate?
From
https://github.com/input-leap/input-leap/blob/master/src/lib/inputleap/mouse_types.h
It appears that we could extend this to mice with a moderately higher number of buttons (hey, at least it's not one of these 15-button Razers ;) ).

Considered issues https://github.com/input-leap/input-leap/issues/51, https://github.com/input-leap/input-leap/pull/242, https://github.com/input-leap/input-leap/issues/594 and https://github.com/input-leap/input-leap/issues/1042, which seem either resolved or distinct.

**Comment by donjan** [comment]
Meanwhile I've happened to also use a 20.04 (Xserver based) client with _barrier_, and the side mouse buttons work!
So the issue seems to be much narrower, and specific to the input-leap - waynergy connection.

**Comment by MYanello** [comment]
Are you using a custom keymap within waynergy? I think this may be the best route to get this to work, but I'm not sure how to define buttons in there, just keypresses.

**Comment by donjan** [comment]
This has been fixed over at [3ef8bc8](https://github.com/r-c-f/waynergy/commit/3ef8bc87a82f30246d47a4df07507a3047f9cda6) and is a non-issue here.
